wandb/integration/yolov8/yolov8.py

```diff
@@ -0,0 +1,266 @@
+                    key,
+                )
+                for key in (1, "inference")
+            ]
+            speed = speeds[0] if speeds[0] else speeds[1]
+            if speed:
+                self.run.summary.update(
+                    {
+                        "model/speed(ms/img)": round(speed, 3),
+                    }
+                )
+        if trainer.best_fitness == trainer.fitness:
+            self.run.summary.update(
+                {
+                    "best/epoch": trainer.epoch + 1,
+                    **{f"best/{key}": val for key, val in trainer.metrics.items()},
+                }
+            )
+
+    def on_train_end(self, trainer: BaseTrainer):
+        """
+        On train end we log all the media, including plots, images and best model artifact to Weights & Biases.
+        """
+        if not isinstance(trainer, ClassificationTrainer):
+            self.run.log(
+                {
+                    "plots": [
+                        wandb.Image(str(image_path), caption=image_path.stem)
+                        for image_path in trainer.save_dir.glob("*.png")
+                    ],
+                    "val_images": [
+                        wandb.Image(str(image_path), caption=image_path.stem)
+                        for image_path in trainer.validator.save_dir.glob("val*.jpg")
+                    ],
+                },
+            )
+
+        if trainer.best.exists():
+            self.run.log_artifact(
+                str(trainer.best),
+                type="model",
+                name=f"{self.run.name}_{trainer.args.task}.pt",
+                aliases=["best", f"epoch_{trainer.epoch + 1}"],
+            )
+
+    def on_model_save(self, trainer: BaseTrainer):
+        """
+        On model save we log the model as an artifact to Weights & Biases.
+        """
+        self.run.log_artifact(
+            str(trainer.last),
+            type="model",
+            name=f"{self.run.name}_{trainer.args.task}.pt",
+            aliases=["last", f"epoch_{trainer.epoch + 1}"],
+        )
+
+    def teardown(self, _trainer: BaseTrainer):
+        """
+        On teardown we finish the Weights & Biases run and set it to None.
+        """
+        self.run.finish()
+        self.run = None
+
+    @property
+    def callbacks(
+        self,
+    ):
+        """
+        Contains all the relevant callbacks to add to the YOLO model for the Weights & Biases logging.
+        """
+        return {
+            "on_pretrain_routine_start": self.on_pretrain_routine_start,
+            "on_pretrain_routine_end": self.on_pretrain_routine_end,
+            "on_train_epoch_start": self.on_train_epoch_start,
+            "on_train_epoch_end": self.on_train_epoch_end,
+            "on_fit_epoch_end": self.on_fit_epoch_end,
+            "on_train_end": self.on_train_end,
+            "on_model_save": self.on_model_save,
+            "teardown": self.teardown,
+        }
+
+
+def add_callbacks(
```

Comment:
This usage is awkward:
```
    model = YOLO("yolov8n.pt")
    model = add_callbacks(model,)
    model.train(data="coco128.yaml", epochs=3, imgsz=640,)
```

The syntax looks like add_callbacks is returning a wrapped object to the passed in object.

Maybe something like:
```
    from wandb.integrations.yolov8 import add_callbacks as add_wandb_callbacks
    model = YOLO("yolov8n.pt")
    add_wandb_callbacks(model, ...)
```

I think the syntax of returning the model just felt strange.

Reply:
:+1: Changed it to reflect the usage mentioned in the comment.